### PR TITLE
Open api docs

### PR DESCRIPTION
**Pull request by ataleksandrov**
- Add versioning of the ORD service APIs
- Swagger docs for REST APIs of the ORD service.
- Add full version in $metadata endpoint of the OData API

### CODE REVIEW

File: components/ord-service/src/main/java/com/sap/cloud/cmp/ord/service/controller/SpecificationsController.java [hunk 1]
```diff
@@ -19,60 +18,96 @@
...
         } catch (IllegalArgumentException e) {
-            handleErrorResponse(response, INVALID_TENANT_ID_ERROR_MESSAGE);
+            respond(response, HttpServletResponse.SC_BAD_REQUEST, e.getMessage());
         }
-        return eventSpec.getSpecData();
     }
 
-    private void handleErrorResponse(HttpServletResponse response, String errorMessage) throws IOException {
-        response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+    private void respond(HttpServletResponse response, int statusCode, String body) throws IOException {
```
Comment (nyordanoff) [inline components/ord-service/src/main/java/com/sap/cloud/cmp/ord/service/controller/SpecificationsController.java:106]: The `throws` declaration is now obsolete

File: components/ord-service/src/main/java/com/sap/cloud/cmp/ord/service/controller/SpecificationsController.java [hunk 2]
```diff
@@ -19,60 +18,96 @@
...
 public class SpecificationsController extends com.sap.cloud.cmp.ord.service.controller.Controller {
 
     private final String NOT_FOUND_MESSAGE = "Not Found";
     private final String INVALID_TENANT_ID_ERROR_MESSAGE = "Missing or invalid tenantID";
 
+    private static final String UNAUTHORIZED_MSG = "{\n" +
+            "  \"error\": {\n" +
+            "    \"code\": 401,\n" +
+            "    \"status\": \"Unauthorized\",\n" +
+            "    \"request\": \"8ed540a8-b4c6-49e3-a5ca-8dc29ba94318\",\n" +
```
Comment (nyordanoff) [inline components/ord-service/src/main/java/com/sap/cloud/cmp/ord/service/controller/SpecificationsController.java:36]: What's the semantic of this hardcoded request id?

Comment (DimitarPetrov) [reply components/ord-service/src/main/java/com/sap/cloud/cmp/ord/service/controller/SpecificationsController.java:36]: It is an example response of an attempt to access the ORD Service without authorization, all the information is a dummy.

**Review by DimitarPetrov — APPROVED** [review]

**Review by nyordanoff — APPROVED** [review]

**Pull request merged**